DESIGN_DOCUMENT_TEMPLATE.md

```diff
@@ -0,0 +1,139 @@
+
+Here is a generic graph diagram you can modify:
+
+```mermaid
+graph TD
+    subgraph Subgraph
+        subgraph Subsubgraph1
+            l1[Label 1]
+            l2[Label 2]
+        end
+        subgraph Subsubgraph2
+            l3(Label 3)
+            l4[Label 4]
+        end
+        subgraph Subsubgraph3
+           l5(Label 5)
+           l6[Label 6]
+        end
+
+        l1 -- Line 1 --> l3
+        l2 -- Line 2 --> l4
+        l3 -- Line 3 --> l5
+    end
+```
+
+
+Similarly, this is a sample sequence diagram:
+
+
+```mermaid
+sequenceDiagram
+	title Sample Sequence
+	autonumber
+	participant A as A
+	participant B as B
+	participant C as C
+	participant D as D
+	participant E as E
+
+    A->>B: Request 1
+    B->>C: Request 1 Received
+    C->>D: Handle Request 
+    D->>C: Do Something
+    C-->>B: Allow/Deny request
+    C->>E: Update Audit Log
+    D-->>B: Result
+    B->>A: Response
+```
+There are several other types of diagrams also supported by Mermaid.
+
+### API Diagram
+
+If your change introduces new API routes, please provide a diagram of the changes.
+
+The sequence diagram is recommended for this purpose:
+
+```mermaid
+sequenceDiagram
+	title Sample Sequence
+	autonumber
+	participant A as A
+	participant B as B
+	participant C as C
+	participant D as D
+	participant E as E
+
+    A->>B: Request 1
+    B->>C: Request 1 Received
+    C->>D: Handle Request 
+    D->>C: Do Something
+    C-->>B: Allow/Deny request
+    C->>E: Update Audit Log
+    D-->>B: Result
+    B->>A: Response
+```
+
+## Documenting your Feature
+
+Whenever you are making a large change, you should make sure you have created the appropriate updates to the documentation website.
+
+Please use this section as a reminder to submit a pull request to the documentation repository with the details of your change.
+
+If you do not complete this step, you pull request may not be merged.
+
+Please link your documentation pull request here when it is complete and check this box:
+- [ ] Documentation pull request opened
+
+## Security Reviews
+
+Certain changes require advanced security reviews by parties which depend on OpenSearch.
+
+As part of this process, we request that the following decision tree is used to determine in which cases an advanced security review is required.
+
+In cases where an advanced review is required, additional time is needed to merge pull requests. In these cases, the earlier pertinent information is provided the better.
+
+Please denote whether a review is expected to be required.
+
+```mermaid
+graph TD
+  A[Is it a new feature or a feature update?] -->|New Feature| B[Does the feature handle sensitive data?]
+  A -->|Feature Update| C[Does the update change security-critical components?]
+  B -->|Yes| D[Open an AppSec review]
+  B -->|No| E[Do a basic security assessment]
+  C -->|Yes| F[Open an AppSec review]
+  C -->|No| G[Proceed with regular testing]
+  E -->|Security issues found?| H[Open an AppSec review]
```

Comment:
Previously released features can have vulnerabilities, which should be reported per our [policy](https://github.com/opensearch-project/security/security/policy).  IMO For features in development we should create issues and link them against the meta issue associated with the feature, less 'process' and high visibility - what do you think?

Reply:
Sorry for which part of the tree did you want removed?

Reply:
Maybe its just how I read `Security issues found?`, I would call this something unrelated to our security policy, or if you did want to include it we should have a clear line between CVEs that are better to address through the security reporting channels, vs design / code feedback during feature development.

Reply:
Gotcha, I switched the arrow tag